recipe_scrapers/_utils.py

```diff
@@ -229,14 +248,9 @@ def change_keys(obj, convert):
     (https://web.archive.org/web/20201022163147/https://stackoverflow.com/questions/11700705/python-recursively-replace
         -character-in-keys-of-nested-dictionary/33668421)
     """
-    if isinstance(obj, (str, int, float)):
-        return obj
     if isinstance(obj, dict):
-        new = obj.__class__()
-        for k, v in obj.items():
-            new[convert(k)] = change_keys(v, convert)
-    elif isinstance(obj, (list, set, tuple)):
-        new = obj.__class__(change_keys(v, convert) for v in obj)
+        return {convert(k): change_keys(v, convert) for k, v in obj.items()}
+    elif isinstance(obj, list):
+        return [change_keys(item, convert) for item in obj]
```

Comment:
Ah, a bit of a nitpick but: what about `set` and `tuple`?  (they are handled in the existing logic)